### PR TITLE
feat: Use GRPC from SPM

### PR DESCRIPTION
**Pull request by AttilaTheFun**
This PR is a port of this PR into RSPM: https://github.com/cgrindel/rules_swift_package_manager/pull/1418

Last year I added support for generating swift_proto_library targets to the Gazelle plugin. 
I added these examples demonstrating the Gazelle plugin in action, but they were limited to the GRPC and SwiftProtobuf versions supplied by rules_swift.

At the time, RSPM was not setting alwayslink = True on the generated swift_library targets which caused the tests to fail on Ubuntu. Since then, this issue has been resolved and the Gazelle plugin has been extracted to this separate repository.

I have ported over my changes to use GRPC and its dependencies from Swift Package Manager. This enables consumers to independently upgrade these dependencies and avoid duplicate symbols with other libraries that also depend on these packages.

### CODE REVIEW

File: examples/grpc_package_example/Package.swift [hunk 1]
```diff
@@ -3,9 +3,11 @@
 import PackageDescription
 
 let package = Package(
-    name: "grpc_package_example",
+    name: "grpc_example",
```
Comment (cgrindel) [inline examples/grpc_package_example/Package.swift:6]: Curious. Why did you rename this? The original name seems better to me.

**Review by cgrindel — APPROVED** [review]
Overall, LGTM. I had one question about the name change in a `Package.swift`. Thank you!

File: examples/grpc_package_example/Package.swift (hunk 1 above)
Comment (AttilaTheFun) [reply examples/grpc_package_example/Package.swift:6]: Ah I copy-pasted the Package.swift file from the other example because the required packages were the same. I can revert the package name.
